File: 621. Task Scheduler.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        # write your code here
        counter = collections.Counter(tasks)
        max_length = max(counter.values())
        offset = 0
        for item in counter:
            if counter[item] == max_length:
                offset += 1
        return max(len(tasks), offset + (max_length - 1) * (n + 1))

File: 621. Task Scheduler.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        unordered_map<char, int> m;
        int max_q = 0;
        for(char task:tasks)
        {
            m[task] += 1;
            max_q = max(m[task], max_q);
        }
        int q = 0;
        for(auto item:m)
        {
            if(item.second == max_q)
                q += 1;
        }
        return max(int(tasks.size()), (max_q - 1) * (n + 1) + q);
    }
};
